crypto/s390xcap.c

```diff
@@ -13,39 +13,52 @@
 #include <setjmp.h>
 #include <signal.h>
 #include "internal/cryptlib.h"
-
-extern unsigned long OPENSSL_s390xcap_P[];
+#include "s390x_arch.h"
 
 static sigjmp_buf ill_jmp;
 static void ill_handler(int sig)
 {
     siglongjmp(ill_jmp, sig);
 }
 
-unsigned long OPENSSL_s390x_facilities(void);
+void OPENSSL_s390x_facilities(void);
+void OPENSSL_vx_probe(void);
 
 void OPENSSL_cpuid_setup(void)
 {
     sigset_t oset;
     struct sigaction ill_act, oact;
 
-    if (OPENSSL_s390xcap_P[0])
+    if (OPENSSL_s390xcap_P.stfle[0])
         return;
 
-    OPENSSL_s390xcap_P[0] = 1UL << (8 * sizeof(unsigned long) - 1);
+    OPENSSL_s390xcap_P.stfle[0] = 1ULL << (sizeof(OPENSSL_s390xcap_P.stfle[0])
+                                           * 8 - 1);
```

Comment:
Just use S390X_CAPBIT(0) here. This was actually initial intention, i.e. to set bit 0 (denoting z/Arch as far as I recall).

Reply:
i didnt get the intention behind this (other than forcing a nonzero value to protect against multiple execution of the constructor by the above "if"), so i left it unmodified. this is overwritten anyway by the stfle instruction , as long stfle returns at least one 64-bit word. If it doesnt, i cant see where the bit is revevant. z/arch mode is S390X_CAPBIT(1).

Reply:
Well, apparently I remember it wrong :-) That zero bit is z/Arch. But intention was to use *a* capability bit, but one that wouldn't actually affect choice of code paths in assembly modules, and bit zero felt kind of appropriate candidate. S390X_CAPBIT(1) would also do, pick either...

Reply:
ah okay, makes sense -- ill stick with capbit 0 but will add a short commentary because the bit choice is not obvious (at least to me :-) )

Reply:
Good thinking!